build failure on power PC

```
devendar@ibm-p9-013 gdrcopy (git::devel)$ make
echo "GDRAPI_ARCH=POWER"
GDRAPI_ARCH=POWER
make: Warning: File `libgdrapi.so.1.2' has modification time 22 s in the future
cc -O2 -fPIC -I /usr/local/cuda/include -I gdrdrv/ -I /usr/local/cuda/include -D GDRAPI_ARCH=POWER  -c -o gdrapi.o gdrapi.c
cc -shared -Wl,-soname,libgdrapi.so.1 -o libgdrapi.so.1.2 gdrapi.o
ldconfig -n /labhome/devendar/gdrcopy
ln -sf libgdrapi.so.1.2 libgdrapi.so.1
ln -sf libgdrapi.so.1 libgdrapi.so
cd gdrdrv; \
make
make[1]: Entering directory `/labhome/devendar/gdrcopy/gdrdrv'
Picking NVIDIA driver sources from NVIDIA_SRC_DIR=/usr/src/nvidia-387.26/nvidia. If that does not meet your expectation, you might have a stale driver still around and that might cause problems.
make[2]: Entering directory `/usr/src/kernels/4.11.0-44.el7a.ppc64le'
make[3]: Warning: File `/labhome/devendar/gdrcopy/gdrdrv/modules.order' has modification time 22 s in the future
make[3]: warning:  Clock skew detected.  Your build may be incomplete.
  Building modules, stage 2.
  MODPOST 2 modules
make[2]: Leaving directory `/usr/src/kernels/4.11.0-44.el7a.ppc64le'
make[1]: Leaving directory `/labhome/devendar/gdrcopy/gdrdrv'
g++ -O2 -I /usr/local/cuda/include -I gdrdrv/ -I /usr/local/cuda/include -D GDRAPI_ARCH=POWER -L /usr/local/cuda/lib64 -L /usr/local/cuda/lib -L /usr/lib64/nvidia -L /usr/lib/nvidia -L /usr/local/cuda/lib64   -o basic basic.o libgdrapi.so.1.2 -lcudart -lcuda -lpthread -ldl
libgdrapi.so.1.2: undefined reference to `_mm_sfence'
collect2: error: ld returned 1 exit status
make: *** [basic] Error 1
```

--- a/gdrapi.c
+++ b/gdrapi.c
@@ -287,12 +287,20 @@ extern int memcpy_cached_store_avx(void *dest, const void *src, size_t n_bytes);
 extern int memcpy_uncached_store_sse(void *dest, const void *src, size_t n_bytes);
 extern int memcpy_cached_store_sse(void *dest, const void *src, size_t n_bytes);
 extern int memcpy_uncached_load_sse41(void *dest, const void *src, size_t n_bytes);
+
+static inline void wc_store_fence(void) { _mm_sfence(); }
+
+#elif defined(GDRAPI_POWER)
+
+static inline void wc_store_fence(void) { asm volatile("sync") ; }
+
 #else // GDRAPI_X86
 static int memcpy_uncached_store_avx(void *dest, const void *src, size_t n_bytes)  { return 1; }
 static int memcpy_cached_store_avx(void *dest, const void *src, size_t n_bytes)  { return 1; }
 static int memcpy_uncached_store_sse(void *dest, const void *src, size_t n_bytes)    { return 1; }
 static int memcpy_cached_store_sse(void *dest, const void *src, size_t n_bytes)    { return 1; }
 static int memcpy_uncached_load_sse41(void *dest, const void *src, size_t n_bytes) { return 1; }
+static inline void wc_store_fence(void) { }
 #endif // GDRAPI_X86
 
 static int first_time = 1;
@@ -357,7 +365,7 @@ int gdr_copy_to_bar(void *bar_ptr, const void *h_ptr, size_t size)
 
         // fencing is needed even for plain memcpy(), due to performance
         // being hit by delayed flushing of WC buffers
-        _mm_sfence();
+        wc_store_fence();
 
     } while (0);
 
@@ -393,7 +401,7 @@ int gdr_copy_from_bar(void *h_ptr, const void *bar_ptr, size_t size)
         memcpy(h_ptr, bar_ptr, size);
 
         // note: fencing is not needed because plain stores are used
-        //_mm_sfence();
+        //wc_store_fence();
 
     } while (0);
 

--- a/gdrapi.c
+++ b/gdrapi.c
@@ -287,21 +287,17 @@ extern int memcpy_cached_store_avx(void *dest, const void *src, size_t n_bytes);
 extern int memcpy_uncached_store_sse(void *dest, const void *src, size_t n_bytes);
 extern int memcpy_cached_store_sse(void *dest, const void *src, size_t n_bytes);
 extern int memcpy_uncached_load_sse41(void *dest, const void *src, size_t n_bytes);
-
 static inline void wc_store_fence(void) { _mm_sfence(); }
+#endif // GDRAPI_X86
 
-#elif defined(GDRAPI_POWER)
-
-static inline void wc_store_fence(void) { asm volatile("sync") ; }
-
-#else // GDRAPI_X86
+#if defined(GDRAPI_POWER)
 static int memcpy_uncached_store_avx(void *dest, const void *src, size_t n_bytes)  { return 1; }
 static int memcpy_cached_store_avx(void *dest, const void *src, size_t n_bytes)  { return 1; }
 static int memcpy_uncached_store_sse(void *dest, const void *src, size_t n_bytes)    { return 1; }
 static int memcpy_cached_store_sse(void *dest, const void *src, size_t n_bytes)    { return 1; }
 static int memcpy_uncached_load_sse41(void *dest, const void *src, size_t n_bytes) { return 1; }
-static inline void wc_store_fence(void) { }
-#endif // GDRAPI_X86
+static inline void wc_store_fence(void) { asm volatile("sync") ; }
+#endif
 
 static int first_time = 1;
 static int has_sse = 0;

--- a/gdrapi.c
+++ b/gdrapi.c
@@ -297,7 +297,7 @@ static int memcpy_uncached_store_sse(void *dest, const void *src, size_t n_bytes
 static int memcpy_cached_store_sse(void *dest, const void *src, size_t n_bytes)    { return 1; }
 static int memcpy_uncached_load_sse41(void *dest, const void *src, size_t n_bytes) { return 1; }
 static inline void wc_store_fence(void) { asm volatile("sync") ; }
-#endif
+#endif // GDRAPI_POWER
 
 static int first_time = 1;
 static int has_sse = 0;

